camel/toolkits/openbb_toolkit.py

```diff
@@ -0,0 +1,1126 @@
+
+        formatted_symbol = self._format_symbol(symbol, asset_type, provider)
+
+        if asset_type == "equity":
+            data = self.client.equity.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "index":
+            data = self.client.index.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "currency":
+            data = self.client.currency.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        elif asset_type == "crypto":
+            data = self.client.crypto.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+        else:
+            data = self.client.equity.price.historical(
+                symbol=formatted_symbol,
+                start_date=start_date,
+                end_date=end_date,
+                interval=interval,
+                provider=provider,
+            )
+
+        return data.to_df() if return_type == "df" else data.results
+
+    def _format_symbol(
+        self, symbol: str, asset_type: str, provider: str
+    ) -> str:
+        """Format symbol based on asset type and provider requirements.
+
+        Args:
+            symbol (str): Original symbol
+            asset_type (str): Type of asset
+            provider (str): Data provider
+
+        Returns:
+            str: Formatted symbol
+        """
+        base_symbol = symbol.replace("=X", "").replace("=F", "")
+        base_symbol = (
+            base_symbol.replace("^", "")
+            .replace("O:", "")
+            .replace("C:", "")
+            .replace("X:", "")
+        )
+
+        if provider == "polygon":
+            if asset_type == "currency":
+                return f"C:{base_symbol}"
+            elif asset_type == "crypto":
+                return f"X:{base_symbol}"
+            elif asset_type == "option":
+                return f"O:{base_symbol}"
+            return base_symbol
+        elif provider == "yfinance":
+            if asset_type == "currency":
+                return f"{base_symbol}=X"
+            elif asset_type == "future":
+                return f"{base_symbol}=F"
+            elif asset_type == "index":
+                return f"^{base_symbol}"
+            return base_symbol
+
+        return base_symbol
+
+    def get_company_fundamentals(
```

Comment:
The openBB supports so many functionalities(https://docs.openbb.co/platform/reference). Only for the fundamentals, it supports lots of options(https://docs.openbb.co/platform/reference/equity/fundamental). Couple general questions:
1. How did we decide which functionalities to integrate first? 
2. Could we do something like `func = self.client.equity.fundamentals.get(statement) func(symbol=symbol, provider=provider)`? So that we don't need to list all the options which are too many. We only need to check if it exists.

Reply:
Wow thanks @willshang76 for your comment, I will think about how to integrate these ideas into the code